spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -59,6 +66,80 @@
     this.batchSize = size;
   }
 
+  protected Map<Integer, ?> findMoreCompoundConstants(Schema tableSchema, Map<Integer, ?> idToConstant) {
+    Map<Integer, Object> compoundTypeConstants = new HashMap<>();
+    List<Types.NestedField> columns = tableSchema.columns();
+
+    for (Types.NestedField nestedField: columns) {
+      compoundTypeConstants = visitCompoundTypesByDFS(nestedField, idToConstant);
+    }
+    compoundTypeConstants.putAll(idToConstant);
+    return compoundTypeConstants;
+  }
+
+  protected boolean containsCompoundType(Schema tableSchema) {
```

Comment:
I don't think we need this. `visitCompoundTypesByDFS` skips primitive types anyway.

Reply:
you are right. i add this for a easy way to view. will remove this soon.